CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ if (CMAKE_CXX_SIMULATE_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_ID STREQUAL "Cl
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR- /arch:AVX /MT /W4 /EHa /EHs")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     message(STATUS "Using Clang/GCC compiler options")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -fpermissive")
```

Comment:
Should we add `-Wall` `-Werror` `-pedantic` for more warnings / catching things? (And the same on MSVC)

Reply:
`-Werror` just breaks everything for no good reason, like the random funcs. Then `-pedantic` I don't see why we would need this flag. `-Wall` gives us warnings which we can/have to ignore, such as it saying `comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’`. I personally don't see why we would need any of these.

Reply:
Its good to have these, but yeah maybe not in this pr. Ill do it myself at some point.